content/articles/oop-in-java-abstraction-and-encapsulation/index.md

```diff
@@ -1,350 +1,482 @@
 
 Now, that we’ve finally gotten abstraction off the list, on to the next one, Encapsulation.
 
-
-#**ENCAPSULATION**
+# ENCAPSULATION
 
 Imagine, you are building an application and the user can directly access and manipulate your data fields, how costly would that be? The outcome is better left to imagine than experience it.
-  
-Apart from Inheritance and Polymorphism which I will write about in my next article and Abstraction, Encapsulation is another fundamental concept of OOP that every OOP developer needs to get familiar with.
 
-Encapsulation is simply the process of encapsulating and hiding the details of a class implementation from the user. Encapsulation is achieved by wrapping the class attributes or fields(variables) and the methods(i.e the codes that manipulate the attributes)together as a unit.
+Apart from Inheritance and Polymorphism which I will write about in my next article. Encapsulation is another fundamental concept of OOP that every developer needs to get familiar with.
+
+Encapsulation is simply the process of encapsulating and hiding the details of a class implementation from the user. Encapsulation is achieved by wrapping the class attributes or variables and the methods together as a unit.
 
 Encapsulation makes for the security of an application as the user of a code cannot access the private variables of an encapsulated class except through its public methods which in most cases are the accessor and mutator methods.
+
 Let’s use a simple code to explain that.
 
 Code snippet
 
 ```java
+
 public class Human{
-	private String name;
-	private int age;
-	private String gender;
+
+   private String name;
+
+   private int age;
+
+   private String gender;
+
 }
+
 ```
 
 From the code snippet above, all fields or attributes of the class are declared private and so the client or in simple terms, the user cannot use or access the class attributes or fields except through the accessor or mutator methods popularly known as ‘Getter’ and ‘Setter’ methods respectively. So, for instance, to set and get the name of an object from the human class, declare the following methods in the Human class.
 
 ```java
+
 public void setName(String name){
-    this.name = name;
+
+   this.name = name;
+
 }
 
 public String getName(){
-	return name;
+
+   return name;
+
 }
-```
 
+```
 
-So because these methods are declared with a public access modifier, other classes can set and get names whenever an object is created from the human class. So encapsulation adds security to our code by reducing how clients access our data or class contract.
+As these methods are public other classes can set and get names whenever an object is created from the human class. Encapsulation secures our code by reducing how clients access our data or class contract.
```

Comment:
> As these methods are public other classes can set and get names whenever an object is created from the human class.

Missing a comma. Also, I would change the word **as** to **since** to be more precise: 

> Since these methods are public, other classes can set and get names whenever an object is created from the human class